futurex_openedx_extensions/helpers/export_data.py

```diff
@@ -0,0 +1,84 @@
+"""
+This module contains a mixin for exporting data to CSV format and related helpers.
+"""
+import csv
+import os
+from datetime import datetime
+from typing import Any, List
+
+from django.conf import settings
+from rest_framework import status as http_status
+from rest_framework.generics import ListAPIView
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+
+class ExportCSVMixin(ListAPIView):
+    """
+    Mixin for exporting data to CSV format.
+    """
+    @property
+    def csv_header(self) -> List[str]:
+        """Get the headers from the serializer fields."""
+        return list(self.get_serializer().fields.keys())
+
+    @property
+    def csv_data(self) -> dict:
+        """Get the serialized data"""
+        queryset = self.get_queryset()
+        serializer = self.get_serializer(queryset, many=True)
+        return serializer.data
+
+    @property
+    def filename(self) -> str:
+        """Get the generated file name with the current timestamp including microseconds"""
+        current_time = datetime.now().strftime('%Y%m%d_%H%M%S_%f')
+        return f'{self.fx_view_name}_{current_time}'
+
+    def build_download_url(self, file: str) -> str:
+        """Return download url from given file"""
+        if file:
+            return self.request.build_absolute_uri(f'/media/{file}')
+        return ''
+
+    def generate_csv_url_response(self) -> dict:
+        """Return response with csv file url"""
+        response_data = {'download_url': '', 'error': ''}
+        try:
+            csv_file = export_data_to_csv(self.csv_data, self.csv_header, self.filename)
+            response_data['download_url'] = self.build_download_url(csv_file)
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            response_data['error'] = f'Failed to write CSV file due to internal error: {str(exc)}'
+        return response_data
+
+    def list(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        """Override the list method to generate CSV and return JSON response with CSV URL"""
+        if self.request.query_params.get('download') == 'csv':
+            response = self.generate_csv_url_response()
+            return Response(response, status=http_status.HTTP_200_OK)
+
+        return super().list(request, args, kwargs)
+
+
+def export_data_to_csv(data: dict, keys: list, filename: str) -> str:
```

Comment:
Thank you @tehreem-sadat , let's now focus on the view mocking without the model and pagination. Most of the work is being done in the mixin which already has access to all the details of the request. What we need is to move the business logic to this independent function. This function should mimic a view request according to the details it receives from the mixin. The details must not be serialized data, they must be the view name, the query parameters, and the`fx_permissions_info`